### PR TITLE
fix(config): add config_path to pipeline provider config keys, and raise ApiError if this happens again

### PR DESCRIPTION
**Pull request by joshuarli**
f/u to https://github.com/getsentry/freight/pull/303. Freight was transparently deleting the config_path I was adding. Now it should raise an ApiError.

### CODE REVIEW

File: freight/providers/utils.py [hunk 1]
```diff
@@ -12,11 +10,16 @@ def parse_provider_config(type, config):
...
     result = {}
-    all_options = chain(
-        list(instance.get_default_options().items()),
-        list(instance.get_options().items()),
-    )
-    for option, option_values in all_options:
+    all_options = {**instance.get_default_options(), **instance.get_options()}
+
+    for key in config:
+        if not all_options.get(key):
```
Comment (asottile-sentry) [inline freight/providers/utils.py:16]: `if key not in all_options` perhaps? -- that way if it's `False` or falsey by default it'll not produce an error?

File: tests/providers/test_shell.py [hunk 2]
```diff
@@ -80,5 +86,6 @@ def test_valid_env(self):
         assert result["env"] == {"FOO": "BAR"}
 
     def test_invalid_env(self):
-        with pytest.raises(ApiError):
+        with pytest.raises(ApiError) as e:
             parse_provider_config("shell", {"command": "/usr/bin/true", "env": "FOO"})
+        assert e.value.message == 'Option "env" is not a valid type for provider: shell'
```
Comment (asottile-sentry) [inline tests/providers/test_shell.py:91]: hmmm what's setting `message`? usually when I see `.message` I assume python 2.x -- typically you'd either use `str(e.value)` or `e.args[0]`

Comment (joshuarli) [reply tests/providers/test_shell.py:91]:
```
class ApiError(Exception):
    def __init__(self, message, name=None, status_code=400):
        self.message = message
```

Comment (joshuarli) [reply tests/providers/test_shell.py:91]: GOnna refactor this, I agree the .message throws me off with python 2.x.

Comment (asottile-sentry) [reply tests/providers/test_shell.py:91]: hah it's also not pickleable -- but maybe that's fine :P

Comment (joshuarli) [reply tests/providers/test_shell.py:91]: Oh, lol, it's already available in __str__.

**Review by asottile-sentry — APPROVED** [review]

**Pull request merged**